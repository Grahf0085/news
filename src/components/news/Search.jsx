import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ term, onChange }) => {

  return <form>
    <input
      id="newsSearch"
      placeholder="Search For News"
      type="text"
      name="newsSearch"
      value={term}
      onChange={onChange}
    />
  </form>;

};

Search.propTypes = {
  term: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Search;
