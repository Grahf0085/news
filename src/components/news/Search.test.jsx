import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('search component', () => {
  it('input box', () => {
    const term = 'olympics';
    const { asFragment } = render(<Search search={term} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
