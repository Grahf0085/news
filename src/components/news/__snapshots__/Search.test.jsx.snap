// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component input box 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="newsSearch"
      name="newsSearch"
      placeholder="Search For News"
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
